fractal-theme/assets/js/src/search.js

```diff
@@ -0,0 +1,109 @@
+import Fuse from 'fuse.js'
+
+function fractalSearch() {
+  const searchElement = document.querySelector('.search');
+  const searchInput = searchElement.querySelector('#searchInput');
+  const searchSuggestions = searchElement.querySelector('#searchSuggestions');
+  const searchExplanation = document.querySelector('.search-color-meaning');
+  const searchExplanationClose = searchExplanation.querySelector('.search-color-meaning__close');
+  const searchForm = document.querySelector('#searchForm');
+  const SearchApi = '/assets/theme/js/search.json';
+
+  // Search options
+  const searchOptions = {
+    keys: [{
+      name: 'data.title',
+      weight: 0.5
+    }, {
+      name: 'data.primaryKeywords',
+      weight: 0.4
+    }, {
+      name: 'data.secondaryKeywords',
+      weight: 0.1
+    }]
+  };
+
+  // Fetch endpoint
+  fetch(SearchApi)
+    .then(response => {
+      return response.json()
+    })
+    // Search init
+    .then(search => {
+      const fuse = new Fuse(search, searchOptions)
```

Comment:
Is it necessarily to create a new instance every time?

Reply:
One could propagate it in an arrow function above and thereby cache the instance perhaps. Or set the library on a variable and then call that variable
e.g. `searchFn = (searchQuery => { fuseLib.search(searchQuery); }`

I would say that the search options are propagated when "newed up" therefore it would make better sense to import or require the "search data" at build time
But i totally understand the want-to/need-to fetch the search results

Reply:
It is two separate instances running at the same time on purpose. This means you can actually search for "links" on the search results page and then also search for "html guidelines" in the navigation.